### PR TITLE
[SPARK-41430][UI] Protobuf serializer for ProcessSummaryWrapper

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
Add Protobuf serializer for ProcessSummaryWrapper

### Why are the changes needed?
Support fast and compact serialization/deserialization for ProcessSummaryWrapper over RocksDB.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
New UT

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala [hunk 1]
```diff
@@ -523,4 +523,32 @@ class KVStoreProtobufSerializerSuite extends SparkFunSuite {
...
+    val result = serializer.deserialize(bytes, classOf[ProcessSummaryWrapper])
+    assert(result.info.id == input.info.id)
+    assert(result.info.hostPort == input.info.hostPort)
+    assert(result.info.isActive == input.info.isActive)
+    assert(result.info.totalCores == input.info.totalCores)
+    assert(result.info.addTime == input.info.addTime)
+    assert(result.info.removeTime == input.info.removeTime)
+    assert(result.info.processLogs.size == input.info.processLogs.size)
+    result.info.processLogs.keys.foreach { k =>
+      assert(result.info.processLogs.contains(k))
```
Comment (gengliangwang) [inline core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala:549]: I don't think we need this check.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
@techaddict can you resolve the conflicts?

**Comment by techaddict** [comment]
@gengliangwang merged, and build passed

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**